update welcome screen, change onboarding swiper colors slightly

housseindjirdeh opened the pull request:
Closes #181. Updates welcome screen so user is automatically navigated when request finishes firing:

![welcome-screen-update](https://user-images.githubusercontent.com/12476932/28747206-8b538bc6-7467-11e7-845a-7fbf735e3293.gif)

Also:
* improved colors used in onboarding swiper
* created a `resetNavigationTo` helper method to simplify things

housseindjirdeh commented:
@andrewda yeah I was thinking the same thing, considered just removing the title message entirely and only having a loading spinner but I think having a minimum timer could be okay as well.

Not sure if it's possible to fade away, agreed that would be nicer. Will investigate.

lex111 commented:
@housseindjirdeh I like the idea with the spinner, so it will be clear.

andrewda commented on src/auth/auth.action.js:
```diff
@@ -21,7 +21,7 @@ export const auth = (code, state, navigation) => {
     dispatch({ type: LOGIN.PENDING });
 
-    fetchAccessToken(code, state)
+    delay(fetchAccessToken(code, state), 2000)
```
Should this be a constant or just keep in hard coded?

housseindjirdeh replied:
Hmm are you thinking of having `2s` as a specific constant like `loadDelay`? Personally feel with the `delay` method in `utils`, might be better to have the freedom to use any number we like if we wish to delay other async actions.

What do you think?

andrewda replied:
Yep, sounds good!

housseindjirdeh commented:
Updated: added a `delay` util method to allow for a _whichever comes first_ promise wrapper.

![welcome-update-2](https://user-images.githubusercontent.com/12476932/28747803-c8891886-7475-11e7-902b-510d2c0312ce.gif)

Unfortunately I don't think [`NavigationActions`](https://reactnavigation.org/docs/navigators/navigation-actions), allows for anything aside from a _stack_ style of navigation so don't know if making it fade would be possible :thinking:

andrewda commented:
@housseindjirdeh Not sure if this would work but: https://github.com/react-community/react-navigation/issues/1490#issuecomment-306009554

housseindjirdeh commented:
Nice I'll definitely give that a shot. Seems to be an iOS fade solution so would probably need to dig deeper to see if that could be applied to Android as well.

Definitely not going to prioritize that today/tomorrow however. I think the way it currently is should suffice for the welcome screen until we get the time to look into integrating fade animations :raised_hands:
